### PR TITLE
UAR-1281 New 'is removal confirmed' field

### PR DESCRIPTION
**Pull request by mwestacott**
https://companieshouse.atlassian.net/browse/UAR-1281

* New model field sent to the API and mapped back when JSON retrieved from the API
* Added test changes and assertions, required to cover the new field

### CODE REVIEW

File: src/services/overseas-entities/mapping.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ export const mapOverseasEntity = (body: OverseasEntity): OverseasEntityResource
         managing_officers_corporate: mapManagingOfficersCorporate(body.managing_officers_corporate),
         trusts: mapTrusts(body.trusts),
         update: mapUpdate(body.update),
-        is_remove: (body.is_remove) ? body.is_remove : null
+        is_remove: (body.is_remove) ? body.is_remove : null,
+        is_remove_not_proprietor_of_land: (body.is_remove_not_proprietor_of_land) ? body.is_remove_not_proprietor_of_land : null
```
Comment (markpit) [inline src/services/overseas-entities/mapping.ts:51]: would it be better to remove the 'not' from the field name, to avoid having not true , or not false.

Comment (markpit) [reply src/services/overseas-entities/mapping.ts:51]: and should the field be optional? as i guess update/register journeys won't populate it

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts:51]: Perhaps. Although I tried to match the field name with the text on the page, so when checked, value is true:

Screenshot 2024-01-17 at 13 55 04

Comment (markpit) [inline src/services/overseas-entities/mapping.ts:51]: the web model has a remove block to store soe remove data, should the new field live under there?

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts:51]: Yes I think it should. Will move it...

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts]: Now moved. Tested with web and API and all seems fine.

**Review by markpit — APPROVED** [review]

**Pull request closed without merging**